### DISCUSSION TITLE
How to select specific fields from a relationship with contains_eager ?

### DISCUSSION THREAD

**Discussion by jossefaz** [sqlalchemy/sqlalchemy#7526] [Usage Questions] [answered]
Hello, 

My problem is simple : I would like to `eager_load` a relationship field but only with some specific fields.

So far I tried this : 
```python
select(Table1, Table2.id).join(Table2).options(contains_eager(Table1.table2))
```
Where Table1 is the parent and Table2 contains the children (one to many).
And `table2` is the relationship field.
But it still getting the whole Table2 object where I only need a list of children ids (This is why I select Table2.id in the `select` method).

How can I achieve this ?

**Comment by jossefaz** [comment, marked as answer]
Okay I found it in the documentation : using the `load_only` method

```python
select(Table1, Table2.id).join(Table2).options(contains_eager(Table1.table2).load_only(Table2.id))
```

**Reply by CaselIT** [reply]
`load_only` works also with the loading strategies supported by sqlalchemy so you can also write the query above like this if you don't want to express the join explicitely
```py
select(Table1).options(joinedload(Table1.table2).load_only(Table2.id))
```
